response.go

```diff
@@ -37,6 +37,7 @@ func (r Response) FrameType() FrameType {
 func (r Response) Write(w *bufio.Writer) (err error) {
 	if err = writeFrameHeader(w, FrameTypeResponse, len(r)); err != nil {
 		err = errors.WithMessage(err, "writing response message")
+		return
```

Comment:
I think this line for error cases when we trying call ` writeFrameHeader` 

if we will remove  this `return`  it will try call `w.WriteString(string(r))` 

I think we should keep it.

Reply:
OK - why do we assign to `err` again here then? We never check the result, it gets overwritten.

Reply:
It is actually does : by calling return.

To be honest  if we will keep this like this  function  will do this:
```
func (r Response) Write(w *bufio.Writer) error {
	if err := writeFrameHeader(w, FrameTypeResponse, len(r)); err != nil {
		return errors.WithMessage(err, "writing response message")
	}
     
        if _, err := w.WriteString(string(r)); err != nil {
	        return errors.Wrap(err, "writing response message")
	}
       
     return nil
}
``` 

If we will remove this line it will be this:

```
func (r Response) Write(w *bufio.Writer) error {
         err := writeFrameHeader(w, FrameTypeResponse, len(r))
	if  err != nil {
		err = errors.WithMessage(err, "writing response message")
	}
     
        if _, err := w.WriteString(string(r)); err != nil {
	        return errors.Wrap(err, "writing response message")
	}
       
     return err
}
```

As I understand :)